h5grove/models.py

```diff
@@ -40,3 +40,54 @@ class LinkResolution(str, Enum):
     },
     total=False,
 )
+
+
+class EntityMetadata(TypedDict):
+    name: str
+    kind: str
+
+
+class ExternalLinkMetadata(EntityMetadata):
+    target_file: str
+    target_path: str
+
+
+class SoftLinkMetadata(EntityMetadata):
+    target_path: str
+
+
+AttributeMetadata = TypedDict(
+    "AttributeMetadata", {"name": str, "shape": tuple, "type": TypeMetadata}
+)
+
+
+class ResolvedEntityMetadata(EntityMetadata):
+    attributes: List[AttributeMetadata]
```

Comment:
```suggestion
    attributes: list[AttributeMetadata]
```

might also need `from __future__ import annotations`

Reply:
To be accurate, it doesn't work for type declaration but only for parameter types.

**Works**

```python
from __future__ import annotations

def my_func(a: list[int]) -> int | None:
```

**Fails**

```
from __future__ import annotations

L = list[int]

def my_func(a: L) -> int | None:
```

In latter case, `List` is needed:

```python
from __future__ import annotations
from typing import List

L = List[int]

def my_func(a: L) -> int | None:
```